Markdown.js

```diff
@@ -30,12 +30,7 @@ sap.ui.define(["sap/ui/core/Control", "./marked/marked.min"], (Control /*, marke
                 } else if (oControl.getFromFile()) {
                     fetch(oControl.getFromFile())
                         .then((r) => r.text())
-                        .then((md) => {
-                            // eslint-disable-next-line no-undef
-                            const sHtml = marked(md)
```

Comment:
but where does the actual markdown to html conversion happen then if the call to `marked` is removed?

Reply:
Okay let me see if I can explain the thing: 
The html conversion happens in the conditional code block above once `sMarkdown` gets truthy. 

When the property `content` is passed to the control, it gets rendered once and we are good to go.

When the property `fromFile` is passed to the control, the file is fetched (async!) in the first rendering cycle. Once the promise resolves, this first rendering cycle has already finished. So we need an other cycle which is initiated by `oControl.setContent(md)` In this second cylce `sMarkdown` can be derived from `oControl.getContent()` an the conversion finally happens.

Reply:
I got one more question, more for myself than anything. ^^

Is there a way to better analyze (visualize maybe) the different rendering cycles in the browser dev tools or which approach did you go with when looking into this - f.e. use several breakpoints & reloads?

Reply:
@phreis: ah, now i get it: you're utilizing an additional rendering cycle triggered by a `$control.setProperty()` in order to make the retrieved markdown available to the renderer logic.
Not sure whether i like this performance-wise but i like the code removal :)

Reply:
First I have to admit that I am still learning UI5  🙂
And yes, your are right I figured out using several breakpoints in chrome dev tool eg. at `if (sMarkdown) {` and in the `.then` branches. 

Further more documentation sais, that `setContent()` marks the control as changed which enforces automatic re-rendering. 
Side note: Invalidation ( and re-rendering) can be suppressed by: `oControl.setProperty('content',md,true`
[https://sapui5.hana.ondemand.com/sdk/#/api/sap.ui.base.ManagedObject%23methods/setProperty](url)